pkg/remote/bound/limiter_inbound_test.go

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2021 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bound
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/cloudwego/kitex/pkg/remote"
+
+	limiterMocks "github.com/cloudwego/kitex/internal/mocks/limiter"
+	"github.com/cloudwego/kitex/internal/test"
+	"github.com/cloudwego/kitex/pkg/kerrors"
+	"github.com/cloudwego/kitex/pkg/remote/trans/invoke"
+)
+
+var errFoo = errors.New("mockError")
+
+// TestNewServerLimiterHandler test NewServerLimiterHandler function and assert result to be not nil.
+func TestNewServerLimiterHandler(t *testing.T) {
+	concurrencyLimiter := &limiterMocks.ConcurrencyLimiter{}
+	rateLimiter := &limiterMocks.RateLimiter{}
+	limitReporter := &limiterMocks.LimitReporter{}
+
+	handler := NewServerLimiterHandler(concurrencyLimiter, rateLimiter, limitReporter)
+	test.Assert(t, handler != nil)
+}
+
+// TestLimiterOnActive test OnActive function of serverLimiterHandler, to assert the concurrencyLimiter 'Acquire' works.
+// If not acquired, the message would be rejected with ErrConnOverLimit error.
+func TestLimiterOnActive(t *testing.T) {
+	t.Run("Test OnActive with limit acquired", func(t *testing.T) {
+		concurrencyLimiter := &limiterMocks.ConcurrencyLimiter{}
+		rateLimiter := &limiterMocks.RateLimiter{}
+		limitReporter := &limiterMocks.LimitReporter{}
+		ctx := context.Background()
+
+		concurrencyLimiter.On("Acquire").Return(true)
+
+		handler := NewServerLimiterHandler(concurrencyLimiter, rateLimiter, limitReporter)
+		ctx, err := handler.OnActive(ctx, invoke.NewMessage(nil, nil))
+
+		test.Assert(t, ctx != nil)
+		test.Assert(t, err == nil)
+	})
+
+	t.Run("Test OnActive with nil reporter", func(t *testing.T) {
```

Comment:
it is suggested to write more detail for the test case: 'Test OnActive with limit acquire false and nil reporter' ; )